.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,45 @@
+name: golangci-lint
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  generate-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v4
+      - id: set-matrix
+        run: |
+          find . -type f -name 'go.mod' -printf '%h\n' \
+          | jq -r -R -s 'split("\n")[:-1] | sort | "matrix=\(.)"' >> $GITHUB_OUTPUT
+
+  golangci:
+    needs: generate-matrix
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
```

Comment:
This ensures that a failure in one sub-module doesn't cancel the ongoing checks of the other sub-modules.